### ISSUE TITLE
Array automatically transforms to object in findById filter

### ISSUE THREAD

**Issue by kamal0808** [strongloop/loopback#3716]
# Description/Steps to reproduce
I'm trying to use the `PersistedModel.findById` function with the following filter..
```
var filter = {
      include: {
        relation: 'productModels',
        scope: {
          fields: ['sku', 'binLocation'],
          where: {
            sku: {
              inq: productSKUsArray
            }
          }
        }
      }
    };
```
where, `productModels` is a hasMany relation with the Persisted Model, 
`sku` and `binLocation` are valid properties of the ProductModel,
and `productSKUsArray` is an array of strings.. `['103564','103561','AnokhiQueenQuilt','jerseystriped']`

When the method is called, I get this error:
```
{ Error: The sku property has invalid clause {"inq":{"[0]":103564,"[1]":103561,"[2]":"AnokhiQueenQuilt","[3]":"jerseystriped"}}
     at HttpInvocation.transformResponse (/apps/workers/node_modules/loopback/node_modules/strong-remoting/lib/http-invocation.js:300:13)
     at Request._callback (/apps/workers/node_modules/loopback/node_modules/strong-remoting/lib/http-invocation.js:266:10)
     at Request.self.callback (/apps/workers/node_modules/loopback/node_modules/strong-remoting/node_modules/request/request.js:186:22)
     at emitTwo (events.js:106:13)
     at Request.emit (events.js:191:7)
     at Request.<anonymous> (/apps/workers/node_modules/loopback/node_modules/strong-remoting/node_modules/request/request.js:1081:10)
     at emitOne (events.js:96:13)
     at Request.emit (events.js:188:7)
     at IncomingMessage.<anonymous> (/apps/workers/node_modules/loopback/node_modules/strong-remoting/node_modules/request/request.js:1001:12)
     at IncomingMessage.g (events.js:292:16)
     at emitNone (events.js:91:20)
     at IncomingMessage.emit (events.js:185:7)
     at endReadableNT (_stream_readable.js:974:12)
     at _combinedTickCallback (internal/process/next_tick.js:80:11)
     at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:128:9) name: 'Error', status: 400, statusCode: 400 }
```

See how it automatically transformed from an array of strings to an object? Why does this happen and how to fix this?

Another thing to keep in mind is that I'm connecting to these models through a remote data-source, so I'm thinking this may have something to do with arrays not being able to transferred in HTTP requests, but I am not sure.

# Link to reproduction sandbox
https://github.com/kamal0808/loopback-sandbox

# Expected result
It should just give the required products in response. It returns all the products if `where.sku.inq` clause is not specified.

# Additional information
```
node -e 'console.log(process.platform, process.arch, process.versions.node)'
linux x64 6.11.1
```
```
npm ls --prod --depth 0 | grep loopback
npm info it worked if it ends with ok
npm info using npm@3.10.10
npm info using node@v6.11.1
+-- loopback@2.18.0
+-- loopback-boot@2.8.1
+-- loopback-component-storage@1.5.1
+-- loopback-datasource-juggler@2.31.0
```

**Comment by jannyHou** [comment]
Hi @kamal0808 thank you for your sandbox repo, I run it and tried with explorer endpoint
GET /StoreConfigModels/{id}
with filter:
`{"include": {"relation": "productModels", "scope": {"fields": ["sku", "binLocation"], "where": {"sku": {"inq": ["103564","103561","AnokhiQueenQuilt","jerseystriped"]}}}}}`

While I could not see the error, without the `where` clause it returns the correct result with only required fields, and with the complete filter(including `where`), it returns the result with an empty array of `productModels`, which is still correct since no item has a `sku`'s value in array ` ["103564","103561","AnokhiQueenQuilt","jerseystriped"]`.

Which connector(remote datasource) are you using?

**Comment by kamal0808** [comment]
@jannyHou I figured, the sandbox env could not reproduce the issue. And I think the main reason for that is running the sandbox in loopback `^v2.22.0`, whereas my original project is running on loopback `v2.18.0`. I'll fix the sandbox to run on the same `v2.18.0` and let you know if I am able to reproduce the issue. 
Question: If it is indeed a loopback version issue, are you going to help me solve it in the same `v2.18.0`? Because moving my project to a newer version of loopback is a huge task, and I'm looking for a short-term solution. If not, I'll just find a workaround/hack for myself for now.

**Comment by jannyHou** [comment]
@kamal0808 I am afraid if the problem is solved in a new minor version, upgrading would be the best solution...If upgrading the whole package has risk, how about I help you figure out which is the fix PR and you cherry-pick it to your local package?
